homeassistant/core.py

```diff
@@ -472,7 +472,8 @@ def set(self, entity_id, new_state, attributes=None):
             old_state = self._states.get(entity_id)
 
             is_existing = old_state is not None
-            same_state = is_existing and old_state.state == new_state
+            same_state = (is_existing and old_state.state == new_state and
+                          not force_update)
```

Comment:
Good catch. Last changed should be updated as we're pretending it's a new state.